deployment.rb

```diff
@@ -415,3 +416,18 @@ def shared_js_dir(*dirs)
 def lib_dir(*dirs)
   deploy_dir('lib', *dirs)
 end
+
+# Since channel ids are derived from user id and other sequential integer ids
+# use a new S3 sources directory for each Test Build to prevent a UI test
+# from inadvertently using a channel id from a previous Test Build.
+# CircleCI environments already suffix the Circle Build number on the sources directory:
+# https://github.com/code-dot-org/code-dot-org/blob/fb53af48ec0598692ed19f340f26d2ed0bd9547b/.circleci/config.yml#L153
+def sources_s3_dir
+  if rack_env?(:production)
+    'sources'
+  elsif rack_env?(:test) && !ENV['CI']
+    "sources_#{rack_env}/#{GitUtils.git_revision_short}"
```

Comment:
I find `!ENV['CI']` confusing, because I also think of our test server as "CI".  Circle also sets a `CIRCLECI` environment variable.  Besides not being locked to a particular CI provider, is there are reason we're using this instead of `ENV['CIRCLECI']`?

Reply:
one observation is that because circle overrides this, we'll get the same behavior if we omit the `&& !ENV['CI']`. I'd suggest removing it for simplicity, and adding a comment explaining that CircleCI overrides it, but I don't feel super strongly about it in case others disagree.

Reply:
I switched to use the more specific `CIRCLECI` environment variable and expanded the comment to explain that just to be safe, we're surpressing this logic in Circle environments